src/Service/FirstRegistrationService.php

```diff
@@ -108,45 +271,170 @@ public function removeSelf(array $object): array
+                // Get the file from the inhoud object.
+                $file = $informatieObject->getValueObject('inhoud')->getFiles()->first();
+            }
+
+            if (isset($file) === false) {
+                continue;
+            }
+
+            $documents['documents'][] = [
+                'title'    => $file->getName(),
+                'filename' => $file->getName(),
+                'content'  => $file->getBase64(),
+            ];
+        }
+
+        // Merge the values and documents array.
+        $mappingArray = array_merge($valuesArray, $documents);
+
+        // Create the dataImport array.
+        $dataImportArray['records'][] = $mappingArray;
+
+        // TODO: Get naam and type of the dataImport.
+        // Create and hydrate the dataImport object.
+        $dataImport = new ObjectEntity($dataImportSchema);
+        $dataImport->hydrate($dataImportArray);
+
+        $this->entityManager->persist($dataImport);
+        $this->entityManager->flush();
+
+        return $this->sendFirstRegistration($dataImport);
+
+    }//end zgwToFirstRegistrationHandler()
+
+
+    /**
+     * A first registration handler that is triggered by an action.
+     *
+     * @param array $data          The data array
+     * @param array $configuration The configuration array
+     *
+     * @return array A handler must ALWAYS return an array
+     */
+    public function sendFirstRegistration(ObjectEntity $object): array
+    {
+        $source = $this->resourceService->getSource($this->configuration['source'], 'common-gateway/first-registration-bundle');
+        $schema = $this->resourceService->getSchema($this->configuration['schema'], 'common-gateway/first-registration-bundle');
+
+        if ($source === null || $schema === null) {
+            return [];
+        }
+
+        $this->logger->debug("EersteInschrijving Object with id {$object->getId()->toString()} was created");
 
         $objectArray = $object->toArray();
         $objectArray = $this->removeSelf($objectArray);
 
         // @TODO $objectArray unset _self etc..
         // Create synchronization.
-        $synchronization = $this->zgwToVrijbrpService->getSynchronization($object, $source, $synchronizationEntity);
+        $synchronization = $this->zgwToVrijbrpService->getSynchronization($object, $source, $schema);
 
         $this->logger->debug("Synchronize (Zaak) Object to: {$source->getLocation()}".$this->configuration['location']);
         // Todo: change synchronize function so it can also push to a source and not only pull from a source:
         // $this->syncService->synchronize($synchronization, $objectArray);
         // Todo: temp way of doing this without updated synchronize() function...
         if ($data = $this->zgwToVrijbrpService->synchronizeTemp($synchronization, $objectArray, $this->configuration['location'])) {
```

Comment:
Didn't we move this synchronizeTemp function to CoreBundle 🤔
shouldn't we use that?
(maybe check for other uses of this function as well)

Reply:
Maybe, if we have more time to test if that keeps working

Reply:
This should be high on your list for cleanup at least, because we moved this to core to prevent duplicate code